modules/options/src/main/scala/scala/build/internal/ScalaNightlyVersion.scala

```diff
@@ -0,0 +1,73 @@
+package scala.build.internal
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker
+import coursier.cache.FileCache
+import coursier.util.{Artifact, Task}
+import dependency.ScalaVersion
+
+import scala.build.EitherCps.{either, value}
+import scala.build.Os
+import scala.build.errors.{BuildException, ScalaVersionError}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+object ScalaNightlyVersion {
+
+  private object Scala2Repo {
+    final case class ScalaVersion(name: String, lastModified: Long)
+    final case class ScalaVersionsMetaData(repo: String, children: List[ScalaVersion])
+
+    val codec: JsonValueCodec[ScalaVersionsMetaData] = JsonCodecMaker.make
+  }
+
+  private def downloadScala2RepoPage(cache: FileCache[Task]): Either[BuildException, Array[Byte]] =
+    either {
+      val scala2NightlyRepo =
+        "https://scala-ci.typesafe.com/ui/api/v1/ui/nativeBrowser/scala-integration/org/scala-lang/scala-compiler"
+      val artifact = Artifact(scala2NightlyRepo).withChanging(true)
+      val res = cache.logger.use {
+        try cache.withTtl(0.seconds).file(artifact).run.unsafeRun()(cache.ec)
+        catch {
+          case NonFatal(e) => throw new Exception(e)
+        }
+      }.left.map { err =>
+        val msg =
+          s"""|Unable to compute the latest Scala 2 nightly version.
+              |Throws error during downloading web page repository for Scala 2.
+              |   Caused by: $err""".stripMargin
+        new ScalaVersionError(msg)
```

Comment:
Hmm, I wonder if shouldn't be a warning and latest local matching scala version should be use.

Generally, now `--S 2.nightly` is unusable without internet access

Reply:
At the first time it is required internet access to download:
```
https://scala-ci.typesafe.com/ui/api/v1/ui/nativeBrowser/scala-integration/org/scala-lang/scala-compiler
```

but, if there is no access to the internet the second time, `coursier` uses previously download web page repository from cache. So, Scala CLI uses the latest local matching scala nightly version.